dask_labextension/manager.py

```diff
@@ -0,0 +1,227 @@
+"""A manager for dask clusters."""
+
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+import functools
+import importlib
+from typing import Any, Callable, Dict, List, Union
+from uuid import uuid4
+
+import dask
+from dask.distributed import Adaptive, Cluster, utils
+
+# A type for a dask cluster model: a serializable
+# representation of information about the cluster.
+ClusterModel = Dict[str, Union[str, int]]
+
+# A type stub for a Dask cluster factory.
+DaskClusterFactory = Callable[[], Cluster]
+
+# Get default cluster factory from configuration
+default_factory_module = importlib.import_module(dask.config.get('labextension.factory.module'))
+default_factory = getattr(default_factory_module, dask.config.get('labextension.factory.class'))
+if dask.config.get('labextension.factory.args') or dask.config.get('labextension.factory.kwargs'):
+    default_factory = functools.partial(default_factory,
+                                        *dask.config.get('labextension.factory.args'),
+                                        **dask.config.get('labextension.factory.kwargs'))
+
+
+class DaskClusterManager:
+    """
+    A class for starting, stopping, and otherwise managing the lifecycle
+    of Dask clusters.
+    """
+
+    def __init__(self, cluster_factory: DaskClusterFactory = default_factory) -> None:
+        """
+        Initialize the cluster manager.
+
+        Parameters
+        ----------
+        cluster_factory : function
+            An optional function that, when called, creates a new
+            Dask cluster for usage. If not given, defaults to a LocalCluster.
+        """
+        self._cluster_factory: DaskClusterFactory = cluster_factory
+        self._clusters: Dict[str, Cluster] = dict()
+        self._adaptives: Dict[str, Adaptive] = dict()
+        self._cluster_names: Dict[str, str] = dict()
+        self._n_clusters = 0
+
+    def start_cluster(self, cluster_id: str = "") -> ClusterModel:
+        """
+        Start a new Dask cluster.
+
+        Parameters
+        ----------
+        cluster_id : string
+            An optional string id for the cluster. If not given, a random id
+            will be chosen.
+
+        Returns
+        cluster_model : a dask cluster model.
+        """
+        if not cluster_id:
+            cluster_id = str(uuid4())
+        cluster = self._cluster_factory()
+        self._n_clusters = self._n_clusters + 1
```

Comment:
```suggestion
        self._n_clusters += 1
```

Reply:
Thanks @dhirschfeld! Too many languages rattling around my head.

Reply:
I think the new suggested changes are fantastic, so I figured I'd use one there. 

It's not much, but I had to do a double take there to make sure it was _just_ adding 1. I thought maybe there's a reason it's not using the short-form - i.e. it was a similarly spelt but different variable on the rhs. Using the augmented assignment operators avoids this ambiguity.